scripts/error-docgen.mjs

```diff
@@ -0,0 +1,242 @@
+						.map((s) => `- ${s.replace('-', '').replaceAll(':', '\\:')}`.trim())
+						.join('\n')}`
+				: undefined,
+			`\n\n`,
+		]
+			.filter((l) => l !== undefined)
+			.join('\n');
+	}
+
+	compilerResult = compilerResult.replace(/https\\?:\/\/docs\.astro\.build\//g, '/');
+	astroResult = astroResult.replace(/https\\?:\/\/docs\.astro\.build\//g, '/');
+
+	fs.writeFileSync(
+		'src/pages/en/reference/error-reference.md',
+		HEADER + compilerResult + astroResult + FOOTER,
+		'utf8'
+	);
+
+	function getMessage(errorName, message, cleanMessage) {
+		if (cleanMessage) {
+			return cleanMessage;
+		}
+
+		if (message) {
+			if (typeof message === 'string') {
+				return `${errorName}: ${message}`;
+			} else {
+				return `${errorName}: ${String.raw({
+					raw: extractStringFromFunction(message.toString()),
+				})}`;
+			}
+		}
+
+		return undefined;
+	}
+}
+
+async function getAstroErrorsData() {
+	const inputBuffer = STUB || (await fetch(errorURL).then((r) => r.text()));
+
+	const compiledResult = ts.transpileModule(inputBuffer, {
+		compilerOptions: { module: 'esnext', target: 'esnext', removeComments: false },
+	}).outputText;
+
+	const encodedJs = encodeURIComponent(compiledResult);
+	const dataUri = 'data:text/javascript;charset=utf-8,' + encodedJs;
+
+	/**
+	 * @type {{AstroErrorData: Object.<string, {code: number, message: string, hint: string}>}
+	 */
+	const data = await import(dataUri);
+
+	const jsDocComments = jsdoc
+		.explainSync({ source: compiledResult })
+		.filter(
+			(data) =>
+				data.kind !== 'package' &&
+				!data.undocumented &&
+				data.tags &&
+				data.tags.some((tag) => tag.title === 'docs')
+		);
+
+	return {
+		errors: data.AstroErrorData,
+		jsdoc: jsDocComments,
+	};
+}
+
+// eslint-disable-next-line no-unused-vars
+async function getCompilerErrors() {
+	const inputBuffer = compilerSTUB || (await fetch(compilerErrorURL).then((r) => r.text()));
+
+	const compiledResult = ts.transpileModule(inputBuffer, {
+		compilerOptions: { module: 'esnext', target: 'esnext', removeComments: false },
+	}).outputText;
+
+	const encodedJs = encodeURIComponent(compiledResult);
+	const dataUri = 'data:text/javascript;charset=utf-8,' + encodedJs;
+
+	/**
+	 * @type {{DiagnosticCode: Object.<string, number>}
+	 */
+	const data = await import(dataUri);
+
+	const jsDocComments = jsdoc
+		.explainSync({ source: compiledResult })
+		.filter(
+			(data) =>
+				data.kind !== 'package' &&
+				!data.undocumented &&
+				data.tags.some((tag) => tag.title === 'docs')
+		);
+
+	return {
+		errors: data.DiagnosticCode,
+		jsdoc: jsDocComments,
+	};
+}
+
+/**
+ * @param {string} func
+ */
+function extractStringFromFunction(func) {
+	const arrowIndex = func.indexOf('=>') + '=>'.length;
+
+	// eslint-disable-next-line no-useless-escape
+	return escapeHtml(func.slice(arrowIndex).trim().slice(1, -1));
+
+	function escapeHtml(unsafe) {
+		return unsafe
+			.replaceAll('${', '`')
+			.replaceAll('}', '`')
+			.replaceAll(':', '\\:')
```

Comment:
Is this addressing the issue we had with colons like `client:only`?

Looks like we also need it for headings:

<img width="856" alt="image" src="https://user-images.githubusercontent.com/357379/202701305-7194867f-bf57-4b6f-becc-799c520d5ac1.png">

Full text should be “Missing hint on client:only directive”.

Reply:
Ah yep, it was to work around the issue. Good catch!